purs-loader generates invalid requires on windows

In particular it produces something like `module.exports = require('.\output\moduleName\fileName.js')` and then webpack generates some cryptic error 'cannot find file "....utputoduleNameileName.js"' (note skipped letters).

Loader should instead generate `'./output/moduleName/fileName.js'` or `'.\\output\\moduleName\\fileName.js'`. Locally I *"fixed"* the bug with ugly hack - added `.replace(/\\/g, '\\\\')` to generated path. And actually I'm not sure how it really should be fixed.

Paths are valid. But it's invalid to use unescaped paths (though perfectly valid) in code. Weird, I know. Well, that's windows.

With your modifications I get:
```
resourceDir: D:\Development\Web\MyBooks.purs\src\Entries\Index
path: ..\..\..\output\bundle.js
```
No problem here, all valid. But then:
```
ERROR in ./src/Entries/Index.purs
Module not found: Error: Cannot resolve module '....outpuundle.js' in D:\Development\Web\MyBooks.purs\src\Entries
 @ ./src/Entries/Index.purs 1:17-50
```

--- a/docs/PursLoader/JsStringEscape.md
+++ b/docs/PursLoader/JsStringEscape.md
@@ -1,0 +1,9 @@
+## Module PursLoader.JsStringEscape
+
+#### `jsStringEscape`
+
+``` purescript
+jsStringEscape :: String -> String
+```
+
+

--- a/package.json
+++ b/package.json
@@ -24,8 +24,7 @@
     "webpack": "^1.8.4"
   },
   "dependencies": {
-    "async": "^1.3.0",
-    "chalk": "^1.1.0",
-    "debug": "^2.2.0"
+    "debug": "^2.2.0",
+    "js-string-escape": "^1.0.1"
   }
 }

--- a/src/PursLoader/JsStringEscape.js
+++ b/src/PursLoader/JsStringEscape.js
@@ -1,0 +1,7 @@
+'use strict';
+
+// module PursLoader.JsStringEscape
+
+var jsStringEscape = require('js-string-escape');
+
+exports.jsStringEscape = jsStringEscape;

--- a/src/PursLoader/JsStringEscape.purs
+++ b/src/PursLoader/JsStringEscape.purs
@@ -1,0 +1,3 @@
+module PursLoader.JsStringEscape (jsStringEscape) where
+
+foreign import jsStringEscape :: String -> String

--- a/src/PursLoader/Loader.purs
+++ b/src/PursLoader/Loader.purs
@@ -21,6 +21,8 @@ import Data.String.Regex (Regex(), match, noFlags, regex)
 
 import Unsafe.Coerce (unsafeCoerce)
 
+import PursLoader.Debug (debug)
+import PursLoader.JsStringEscape (jsStringEscape)
 import PursLoader.LoaderRef
   ( AsyncCallback()
   , LoaderRef()
@@ -30,8 +32,6 @@ import PursLoader.LoaderRef
   , addDependency
   , resourcePath
   )
-
-import PursLoader.Debug (debug)
 import PursLoader.Path (dirname, joinPath, relative)
 import PursLoader.Plugin as Plugin
 
@@ -79,13 +79,13 @@ loader ref source = do
          else moduleExport <<< modulePath <$> name
       where
       bundleExport :: String -> String
-      bundleExport name' = "module.exports = require('" ++ path ++ "')['" ++ name' ++ "'];"
+      bundleExport name' = "module.exports = require('" ++ jsStringEscape path ++ "')['" ++ name' ++ "'];"
         where
         path :: String
         path = relative resourceDir pluginContext.options.bundleOutput
 
       moduleExport :: String -> String
-      moduleExport path = "module.exports = require('" ++ path ++ "');"
+      moduleExport path = "module.exports = require('" ++ jsStringEscape path ++ "');"
 
       modulePath :: String -> String
       modulePath = relative resourceDir <<< joinPath pluginContext.options.output

